core/include/detray/core/detector.hpp

```diff
@@ -634,56 +637,34 @@ struct detector_buffer {
-          _bfield_view(det.get_bfield()) {}
+          _bfield(det.get_bfield()) {}
 
     /// Buffers were created manually
     detector_buffer(
-        detector<metadata, bfield_t, container_t> &det,
-        detail::get_buffer_t<typename detector_type::volume_container>
+        const detector_t &det,
+        detail::get_buffer_t<typename detector_t::volume_container>
             &&vol_buffer,
-        typename detector_type::transform_container::buffer_type &&trf_buffer,
-        typename detector_type::mask_container::buffer_type &&msk_buffer,
-        typename detector_type::material_container::buffer_type &&mat_buffer,
-        typename detector_type::surface_container::buffer_type &&sf_buffer,
-        typename detector_type::volume_finder::buffer_type &&vgrd_buffer)
+        typename detector_t::transform_container::buffer_type &&trf_buffer,
+        typename detector_t::mask_container::buffer_type &&msk_buffer,
+        typename detector_t::material_container::buffer_type &&mat_buffer,
+        typename detector_t::surface_container::buffer_type &&sf_buffer,
+        typename detector_t::volume_finder::buffer_type &&vgrd_buffer)
         : _detector_buffer(std::move(vol_buffer), std::move(trf_buffer),
                            std::move(msk_buffer), std::move(mat_buffer),
                            std::move(sf_buffer), std::move(vgrd_buffer)),
-          _bfield_view(det.get_bfield()) {}
+          _bfield(det.get_bfield()) {}
 
     /// Buffers for the vecemem types
-    typename detector_type::buffer_type _detector_buffer;
+    typename detector_t::buffer_t _detector_buffer;
     /// Covfie field
-    typename detector_type::bfield_type::view_t _bfield_view;
+    bfield_t _bfield;
 };
 
-// Deduction guide to construct a detector buffer type more conveniently
-template <typename metadata, template <typename> class bfield_t,
-          typename container_t>
-detector_buffer(
-    detector<metadata, bfield_t, container_t> &,
-    detail::get_buffer_t<typename detector<metadata, bfield_t,
-                                           container_t>::volume_container> &&,
-    typename detector<metadata, bfield_t,
-                      container_t>::transform_container::buffer_type &&,
-    typename detector<metadata, bfield_t,
-                      container_t>::mask_container::buffer_type &&,
-    typename detector<metadata, bfield_t,
-                      container_t>::material_container::buffer_type &&,
-    typename detector<metadata, bfield_t,
-                      container_t>::surface_container::buffer_type &&,
-    typename detector<metadata, bfield_t,
-                      container_t>::volume_finder::buffer_type &&)
-    -> detector_buffer<metadata, bfield_t, container_t>;
-
 /// @brief A static inplementation of detector data for device
-template <typename metadata, template <typename> class bfield_t,
-          typename container_t>
+template <typename bfield_t, typename detector_t>
```

Comment:
Why is the ```detector_t``` template needed? I would rather use as few templates as possible here

Reply:
I still need some template argument for defining all of the data views. While before this was templated on the `metadata` and `container_types`, I just changed it to `detector_t` since I thought it would be simpler. 
I could revert this change but I still think this is a simpler setup.